Add test coverage (#1)
kanopy-platform/drone-validation: README.md

> # Validation extension for Drone

Reviewer: Some kind of description about what kinds of problems this tool solves, how it works with OPA, etc. would be nice.

Author: Added some details about the implementation and use cases.

# Validation extension for Drone

_Requires Drone server version 1.4 or higher._

This is a simple validation extension that performs pipeline type validations.

When a user prepared a Drone configuration file using a pipeline that was not supported by our platform, the build used to hang and users did not receive any error or feedback. Whith this validation extension enabled, the same build would instantly fail and an error message returned to the UI informing the reason why the build has failed.

## Implementation details

The validation process relies on an [OPA policy](https://www.openpolicyagent.org/) written in [rego](https://www.openpolicyagent.org/docs/latest/policy-language/) and is integrated via [OPA Go API](https://www.openpolicyagent.org/docs/latest/integration/#integrating-with-the-go-api).

Although the main purpose of this extension's policy is to validate supported pipeline types, it can be easily modified to check different attributes of the Drone configuration, including secrets.

> We are currently relying on a simple rego query that requires the policy to return a boolean value on the path `data.drone.validation.deny` and an error message on `data.drone.validation.out`.

Reviewer: ```suggestion
This application runs a simple rego query that requires the policy to return a boolean value on the path `data.drone.validation.deny` and an error message on `data.drone.validation.out`.
```

> The current policy (`validation.rego`) is based on the following workflow:

Reviewer: ```suggestion
The [default policy](policy/validation.rego) is based on the following workflow:
```
